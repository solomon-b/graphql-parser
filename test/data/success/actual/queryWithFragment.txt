[ SelField
    ( Field
        { fieldAlias = Just "leftComparison"
        , fieldName = "hero"
        , fieldDirectives = []
        , fieldArguments = fromList
            [
                ( "episode"
                , VString "EMPIRE"
                )
            ]
        , fieldSelectionSet =
            [ SelFragSpread
                ( FragmentSpread
                    { _fsName = "comparisonFields"
                    , _fsDirectives = []
                    }
                )
            ]
        }
    )
, SelField
    ( Field
        { fieldAlias = Just "rightComparison"
        , fieldName = "hero"
        , fieldDirectives = []
        , fieldArguments = fromList
            [
                ( "episode"
                , VString "JEDI"
                )
            ]
        , fieldSelectionSet =
            [ SelFragSpread
                ( FragmentSpread
                    { _fsName = "comparisonFields"
                    , _fsDirectives = []
                    }
                )
            ]
        }
    )
]