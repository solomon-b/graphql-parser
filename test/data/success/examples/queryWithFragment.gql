{
  leftComparison: hero(episode: "EMPIRE") {
    ...comparisonFields
  }
  rightComparison: hero(episode: "JEDI") {
    ...comparisonFields
  }
}
