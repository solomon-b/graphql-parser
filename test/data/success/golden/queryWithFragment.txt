Left
    ( Left
        ( Field
            { fieldAlias = Just "leftComparison"
            , fieldName = "hero"
            , fieldArguments = fromList
                [
                    ( "episode"
                    , VString "EMPIRE"
                    )
                ]
            , fieldDirectives = []
            , fieldSelectionSet = Just
                ( Left
                    ( Right
                        ( FragmentSpread
                            { fsName = "comparisonFields"
                            , fsDirectives = []
                            }
                        )
                    ) :| []
                )
            }
        )
    ) :|
    [ Left
        ( Left
            ( Field
                { fieldAlias = Just "rightComparison"
                , fieldName = "hero"
                , fieldArguments = fromList
                    [
                        ( "episode"
                        , VString "JEDI"
                        )
                    ]
                , fieldDirectives = []
                , fieldSelectionSet = Just
                    ( Left
                        ( Right
                            ( FragmentSpread
                                { fsName = "comparisonFields"
                                , fsDirectives = []
                                }
                            )
                        ) :| []
                    )
                }
            )
        )
    ]